src/renderers/shared/stack/reconciler/ReactUpdateQueue.js

```diff
@@ -244,6 +268,15 @@ var ReactUpdateQueue = {
       (internalInstance._pendingStateQueue = []);
     queue.push(partialState);
 
+    if (callback) {
+      ReactUpdateQueue.validateCallback(callback, callerName);
+      if (internalInstance._pendingCallbacks) {
+        internalInstance._pendingCallbacks.push(callback);
+      } else {
+        internalInstance._pendingCallbacks = [callback];
+      }
+    }
```

Comment:
Do we manually inline these 3 blocks (and the one in `enqueueCallback`) for performance reasons?